c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+
+	switch (UserChoice)
+	{
+	case MainMenuCommand::AddGameMenu:
+		ActiveCommand = MenuCommand::AddGamesMenu;
+		break;
+
+	case MainMenuCommand::CategoryManagerMenu:
+		ActiveCommand = MenuCommand::CategoryManagerMenu;
+		break;
+
+	case MainMenuCommand::GameDisplayerMenu:
+		ActiveCommand = MenuCommand::GameDisplayerMenu;
+		break;
+
+	case MainMenuCommand::Exit:
+		ActiveCommand = MenuCommand::Exit;
+		break;
+
+	default:
+		std::cout << "MainMenu Error, please contact support";
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenAddGamesMenu()
+{
+	system("cls");
+
+	// CategoryIndex is set to -1 and that value is representing the uncategorized category
+	int CategoryIndex = -1;
+
+	if (!CategoryContainer.IsEmpty())
+	{
+		CategoryIndex = ChooseCategory();
+		system("cls");
+	}
+
+	std::string GameName;
+	std::cout << "Please insert the name of the game: ";
+	std::cin >> GameName;
+	GameName = ValidateInputAndReturnValidatedInput(GameName, "Please insert a valid name for the game: ");
+
+	std::string GameStudio;
+	std::cout << "Please insert the name of the game studio: ";
+	std::cin >> GameStudio;
+	GameStudio = ValidateInputAndReturnValidatedInput(GameStudio, "Please insert a valid name for the game studio: ");
+
+	float TemporalYear;
+	std::cout << "Please insert the year the game was published: ";
+	std::cin >> TemporalYear;
+	const int Year = ValidateInputAndReturnValidatedInput(TemporalYear, "Please insert the actual year the game was published: ", 1900, 2022);
+
+	float TemporalMonth;
+	std::cout << "Please insert the month the game was published (1-12): ";
+	std::cin >> TemporalMonth;
+	int Month = ValidateInputAndReturnValidatedInput(TemporalMonth, "Please insert the actual month the game was published (1-12): ", 1, 12);
+
+	float TemporalDay;
+	std::cout << "Please insert the day the game was published: ";
+	std::cin >> TemporalDay;
+	int Day = GetValidDay(TemporalDay, Month, Year);
+
+	FVideogame Game = FVideogame(GameName, GameStudio, FDate(Day, Month, Year));
+
+	if (CategoryIndex == -1)
+	{
+		Uncategorized.AddGame(Game);
+	}
+
+	else
+	{
+		CategoryContainer.AddGameToCategory(CategoryIndex, Game);
+	}
+
+	NumberOfGames++;
+
+	ActiveCommand = MenuCommand::MainMenu;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenCategoryManagerMenu()
+{
+	system("cls");
+	std::string CategoryManagerMessage = "What would you rather do next? \n1 - Add a new category\n2 - Remove a category\n3 - Go back to main menu\nTELL ME : ";
+	std::cout << CategoryManagerMessage;
+
+	enum class CategoryManagerCommand { CreateCategory = 1, RemoveCategory, MainMenu };
```

Comment:
according to the coding standards, the enums have to be named with the prefix `E`